moveit_core/CMakeLists.txt

```diff
@@ -51,8 +51,12 @@ else()
 endif()
 
 find_package(octomap REQUIRED)
+
 find_package(ruckig REQUIRED)
-set(ruckig_LIBRARIES "ruckig::ruckig")
+get_target_property(ruckig_LOCATION ruckig::ruckig LOCATION)
+get_target_property(ruckig_INCLUDE_DIRS ruckig::ruckig INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
I think this is the only line that needs to be added to correctly set include paths.
In Noetic the includes are just found via `/opt/ros/*/include`.

Reply:
Technically that makes it compile, but it doesn't resolve the `ruckig::ruckig` target which will still be added to the `DEPENDS`, resulting in potential errors in downstream packages.

That might still happen because ruckig requires c++17, but at least I propose to make it explicit here.

Reply:
> Technically that makes it compile, but it doesn't resolve the `ruckig::ruckig` target which will still be added to the `DEPENDS`, resulting in potential errors in downstream packages.

cmake correctly resolves library targets to the corresponding library via [`catkin_replace_imported_library_targets`](https://github.com/ros/catkin/blob/bb2299bc878a638d5a81226c21cb918359e773fc/cmake/catkin_package.cmake)
However, you are right that includes are not yet resolved.

Reply:
Wow, now that's a(n incomplete/broken) feature I did not know. :+1: 

The c++17 declaration is also imported in `moveit_core`, I guess.
In [`moveit_ros_planning` it is set by the library user](https://github.com/ros-planning/moveit/blob/6ac1b9fcaee8f676d2f2758805d5f4d76415aa1d/moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt#L20) which should not be done :disappointed: .
I will move the version bump to a different PR though.